### PR TITLE
Normalize `ThemeData.cardTheme`

### PR DESCRIPTION
**Pull request by QuncCccccc**
Following https://github.com/flutter/flutter/pull/151914, this PR is to normalize `ThemeData.cardTheme`; change the `CardTheme cardTheme` property to `CardThemeData cardTheme` in `ThemeData`. In `ThemeData()` and `ThemeData.copyWith()`, the `cardTheme` parameter type is changed to `Object?` to accept both `CardTheme` and `CardThemeData` so that we won't cause immediate breaking change and make sure rolling is smooth. Once all component themes are normalized, these `Object?` types should be changed to `xxxThemeData`.

There's no way to create a dart fix because we can't add a "@deprecated" label for `CardTheme` because `CardTheme` is a new InheritedWidget subclass now.

Addresses the "theme normalization" sub project within https://github.com/flutter/flutter/issues/91772

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Comment by QuncCccccc** [comment]
Will wait until rest of the component themes(DialogTheme, InputDecorationTheme and TabBarTheme) have fixes ready to merge because I think it might be better to include them together in one breaking change migration.

**Review by guidezpl — APPROVED** [review]
Nice!

File: packages/flutter/test/material/card_theme_test.dart [hunk 1]
```diff
@@ -220,6 +220,121 @@ void main() {
...
+    expect(material.clipBehavior, clipBehavior);
+    expect(material.color, color);
+    expect(material.shadowColor, shadowColor);
+    expect(material.elevation, elevation);
+    expect(material.shape, shape);
+    expect(cardMargin.padding, margin);
+  });
+
+
+  testWidgets('Loacal CardTheme can override global CardTheme', (WidgetTester tester) async {
```
Comment (bleroux) [inline packages/flutter/test/material/card_theme_test.dart:279]:
```suggestion
  testWidgets('Local CardTheme can override global CardTheme', (WidgetTester tester) async {
```

File: packages/flutter/test/material/card_theme_test.dart [hunk 2]
```diff
@@ -220,6 +220,121 @@ void main() {
...
+            child: Card(
+              child: SizedBox(
+                width: 200,
+                height: 200,
+              ),
+            ),
+          ),
+        ),
+      )
+     );
```
Comment (bleroux) [inline packages/flutter/test/material/card_theme_test.dart:325]:
```suggestion
    await tester.pumpWidget(MaterialApp(
      theme: themeData,
      home: const Scaffold(
        body: CardTheme(
          data: CardThemeData(
            clipBehavior: localClipBehavior,
            color: localColor,
            shadowColor: localShadowColor,
            elevation: localElevation,
            margin: localMargin,
            shape: localShape,
          ),
          child: Card(
            child: SizedBox(
              width: 200,
              height: 200,
            ),
          ),
        ),
      ),
    ));
```

File: packages/flutter/test/material/card_theme_test.dart (hunk 1 above)
Comment (bleroux) [inline packages/flutter/test/material/card_theme_test.dart:279]:
```suggestion
  });

  testWidgets('Loacal CardTheme can override global CardTheme', (WidgetTester tester) async {
```

**Review by bleroux — COMMENTED** [review]
While reading for learning purpose, I spotted some typos.

**Comment by QuncCccccc** [comment]
> While reading for learning purpose, I spotted some typos.

Thanks a lot for the suggestions:) Just updated!

File: packages/flutter/lib/src/material/theme_data.dart [hunk 3]
```diff
@@ -315,7 +315,8 @@ class ThemeData with Diagnosticable {
     BottomNavigationBarThemeData? bottomNavigationBarTheme,
     BottomSheetThemeData? bottomSheetTheme,
     ButtonThemeData? buttonTheme,
-    CardTheme? cardTheme,
+    // TODO(QuncCccccc): Change the parameter type to CardThemeData
+    Object? cardTheme,
```
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/theme_data.dart:319]: Similar to #155129, to help smooth rolling into G3, I updated the parameter type in `ThemeData()` and `ThemeData.copyWith()` to `object?` and added a TODO to change the parameter type back to `CardThemeData`.

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

**Pull request merged**